src/Shared/TaskHostConfiguration.cs

```diff
@@ -385,6 +400,13 @@ public void Translate(ITranslator translator)
                                  collectionFactory: count => new HashSet<string>());
 #else
                                  collectionFactory: count => new HashSet<string>(count, StringComparer.OrdinalIgnoreCase));
+#endif
+            translator.Translate(collection: ref _warningsAsMessages,
+                                 objectTranslator: (ITranslator t, ref string s) => t.Translate(ref s),
+#if CLR2COMPATIBILITY
+                                 collectionFactory: count => new HashSet<string>());
```

Comment:
I see that this is the same as `_warningsAsErrors` a few lines above but I'm wondering if we shouldn't be constructing with `StringComparer.OrdinalIgnoreCase` on CLR2 as well. The constructor taking both capacity and comparer was introduced later but the one taking `IEqualityComparer<T>` should be available.